gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -0,0 +1,114 @@
+	fi
+	exit 1
+}
+
+#Set Debug SSID
+if [ $(uci get gluon.debug_ssid_enabled||echo false) = 'true' ] ; then
+	suffix="$(uci get network.bat0.macaddr)"
+	if ! grep -q "dev.ffmuc.net/$suffix" /etc/config/wireless ; then
+		sed "s|muenchen.freifunk.net/.*|dev.ffmuc.net/$suffix'|g" /etc/config/wireless -i
+		uci -q set wireless.radio0.txpower="$(uci -q get gluon.debug_tx_power|| echo 0)"
+		uci -q set wireless.radio1.txpower="$(uci -q get gluon.debug_tx_power|| echo 0)"
+		wifi
+	fi
+fi
+
+# Do we already have a private-key? If not generate one
+temp=$(uci get wireguard.mesh_vpn.privatekey)
+if [ "$?" -ne "0" ]; then
+	uci set wireguard.mesh_vpn.privatekey=$(wg genkey)
+	uci commit wireguard
+fi
+
+# Is wireguard enabled?
+if [ "$(uci get wireguard.mesh_vpn.enabled)" == "true" ] || [ "$(uci get wireguard.mesh_vpn.enabled)" == "1" ]; then
+
+	#We assume we are not connected by default
+	CONNECTED=0
+
+	MESH_VPN_IFACE=$(uci get wireguard.mesh_vpn.iface)
+
+	# Check connectivity to supernode
+	wget http://[$(wg  | grep fe80 | awk '{split($3,A,"/")};{print A[1]}')%$MESH_VPN_IFACE]/  --timeout=5 -O/dev/null 
+	if [ "$?" -eq "0" ]; then
+			CONNECTED=1
+	fi
+
+	# If we don't have a connection we try to connect
+	if [ "$CONNECTED" -ne "1" ]; then
+		logger -t checkuplink "Reconnecting ..."
+		NTP_SERVER=$(uci get system.ntp.server)
+		gluon-wan /usr/sbin/ntpd -n -N -S /usr/sbin/ntpd-hotplug -p $NTP_SERVER -q
+
+		# Get the number of configured peers and randomly select one
+		NUMBER_OF_PEERS=$(uci -q show wireguard | egrep -ce peer_[0-9]+.endpoint)
+		PEER="$(awk -v min=1 -v max=$NUMBER_OF_PEERS 'BEGIN{srand(); print int(min+rand()*(max-min+1))}')"
+		PEER_PUBLICKEY="$(uci get wireguard.peer_$PEER.publickey)"
+
+		logger -t checkuplink "Selected peer $PEER"
+
+		endpoint="$(check_handshake "$PEER_PUBLICKEY" "$(uci get wireguard.peer_$PEER.endpoint)")"
+
+		logger -t checkuplink "Connecting to $endpoint"
+
+		# Delete Interfaces
+		ip link set nomaster bat0 dev mesh-vpn &> /dev/null
+		ip link delete dev mesh-vpn &> /dev/null
+		ip link del $MESH_VPN_IFACE &> /dev/null
+		PUBLICKEY=$(uci get wireguard.mesh_vpn.privatekey | wg pubkey)
+		SEGMENT=$(uci get gluon.core.domain)
+
+		# Push public key to broker
+		gluon-wan wget -q  -O- --post-data='{"domain": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' http://broker.ffmuc.net/api/v1/wg/key/exchange
+
+		# Bring up the wireguard interface
+		ip link add dev $MESH_VPN_IFACE type wireguard
+		uci get wireguard.mesh_vpn.privatekey | wg set $MESH_VPN_IFACE private-key /proc/self/fd/0
+		ip link set up dev $MESH_VPN_IFACE
+
+		# Add link-address and Peer
+		ip address add "$(interface_linklocal "$MESH_VPN_IFACE")"/64 dev $MESH_VPN_IFACE
+		if [ "$endpoint" == "" ]; then
+			endpoint=$(uci get wireguard.peer_$PEER.endpoint)
+		fi
+		gluon-wan wg set $MESH_VPN_IFACE peer $(uci get wireguard.peer_$PEER.publickey) persistent-keepalive 25 allowed-ips $(uci get wireguard.peer_$PEER.link_address)/128 endpoint $endpoint
```

Comment:
I'm not sure gluon-wan affects this call at all. We should probably make wireguard set fwmark.